File: DemoServer/Indexes/ProductsAndPriceAndSuplier.cs
using System.Linq;
using DemoServer.Entities;
using Raven.Client.Indexes;

namespace DemoServer.Indexes
{
    public class ProductsAndPriceAndSuplier : AbstractIndexCreationTask<Product>
    {
        public class Result
        {
            public string ProductId { get; set; }
            public decimal PricePerUnit { get; set; }
            public int UnitsInStock { get; set; }
        }

        public ProductsAndPriceAndSuplier()
        {
            Map = products => from product in products
                              select new Result
                              {
                                  ProductId = product.Id,
                                  PricePerUnit = product.PricePerUnit,
                                  UnitsInStock = product.UnitsInStock
                              };
        }

    }
}

File: DemoServer/JavaDemo/src/main/java/net/ravendb/demo/indexes/ProductsAndPriceAndSuplier.java
package net.ravendb.demo.indexes;

import net.ravendb.client.documents.indexes.AbstractIndexCreationTask;

public class ProductsAndPriceAndSuplier extends AbstractIndexCreationTask {

    public static class Result {

        private String productId;
        private double pricePerUnit;
        private int unitsInStock;

        public String getProductId() {
            return productId;
        }

        public void setProductId(String productId) {
            this.productId = productId;
        }

        public double getPricePerUnit() {
            return pricePerUnit;
        }

        public void setPricePerUnit(double pricePerUnit) {
            this.pricePerUnit = pricePerUnit;
        }

        public int getUnitsInStock() {
            return unitsInStock;
        }

        public void setUnitsInStock(int unitsInStock) {
            this.unitsInStock = unitsInStock;
        }
    }

    public ProductsAndPriceAndSuplier() {
        map = "from product in docs.Products " +
                "select new { " +
                "  ProductId = product.__document_id, " +
                "  PricePerUnit = product.PricePerUnit, " +
                "  UnitsInStock = product.UnitsInStock " +
                "}";
    }

}
